perceval/backends/core/rocketchat.py

```diff
@@ -0,0 +1,338 @@
+        self.roomId = roomId
+        self.user_id = user_id
+
+    def room_info(self, roomId):
+        """Fetch information about a room."""
+
+        resource = self.ROOM_INFO
+
+        params = {
+            self.ROOM_ID: roomId,
+        }
+
+        response = self._fetch(resource, params)
+
+        return response
+
+    def history(self, roomId, oldest=None):
+        """Fetch the history of a room."""
+
+        resource = self.ROOM_HISTORY
+
+        params = {
+            self.ROOM_ID: roomId,
+            self.PCOUNT: self.max_items,
+            self.POLDEST: "oldest",
+        }
+
+        response = self._fetch(resource, params)
+
+        return response
+
+    def _fetch(self, resource, params):
+        """Fetch a resource.
+        :param resource: resource to get
+        :param params: dict with the HTTP parameters needed to get
+            the given resource
+        """
+        url = urijoin(self.base_url, API_EXTENSION, resource)
+
+        headers = {
+            self.AUTH_TOKEN: self.api_token,
+            self.USER_ID: self.user_id
+        }
+
+        logger.debug("RocketChat client requests: %s params: %s",
+                     resource, str(params))
+
+        r = self.fetch(url, payload=params, headers=headers)
+
+        # Check for possible API errors
+        result = r.json()
+
+        if r.status_code != 200:
+            raise RocketChatClientError(error=result['error'])
+
+        return r.text
+
+
+class RocketChatClientError(BaseError):
+    """Raised when an error occurs using the RocketChat client"""
+
+    message = "%(error)s"
+
+
+class RocketChatCommand(BackendCommand):
+    """Class to run RocketChat backend from the command line."""
+
+    BACKEND = RocketChat
+
+    @classmethod
+    def setup_cmd_parser(cls):
+        """Returns the Rocket Chat argument parser."""
+
+        parser = BackendCommandArgumentParser(cls.BACKEND,
+                                              from_date=True,
+                                              token_auth=True,
+                                              archive=True,
+                                              ssl_verify=True)
+
+        # Backend token is required
+        token = parser.parser._option_string_actions['--api-token']
+        token.required = True
+
+        # Required arguments
+        parser.parser.add_argument('user_id',
+                                   help="User identifier to fetch messages")
+
+        # Required arguments
+        parser.parser.add_argument('url',
+                                   help="URL of the RocketChat server")
+
+        # Rocket Chat options
+        group = parser.parser.add_argument_group('Rocket.chat arguments')
+        group.add_argument('--max-items', dest='max_items',
+                           type=int, default=MAX_ITEMS,
+                           help="Maximum number of items requested on the same query")
+        group.add_argument('--room-id', dest='roomId',
```

Comment:
Moved it to required! Also, there was only one argument remaining in the argument group `rocketchat arguments`, and I decided to keep the group instead of removing it even though it has only one argument for the sake of consistency and in the case that other arguments need to be added in the future.